### ISSUE TITLE
Enhanced 'MenuButton' component

### ISSUE THREAD

**Issue by lbwexler** [xh/hoist-react#2230]
I was unable to figure out how to use the  MenuButton component.  It seems to have a very different API than what I would expect for a wrapped up control.  In particular, it requires a Model be provided, rather just dropping in data for its items.

I understand that our Menu API is extremely mixed, but it would be nice if this control were a little simpler to drop in.

(Also could use a toolbox example of this)

**Comment by lbwexler** [comment]
Update -- I see now that I need to actually place and render the menu myself.  So the button really doesn't give me too much, other than a consistent way to render itself, and the little click handler to open the menu!

**Comment by TomTirapani** [comment]
The comment kind of says it all:
```
Convenience Button preconfigured for use as a trigger for a dropdown menu operation.
```
We have a bunch of 'preconfigured' buttons like this, across desktop and mobile, which are simply there to offer a consistent way to render and a click handler (e.g. FeedbackButton, OptionsButton, LogoutButton)

**Comment by lbwexler** [comment]
Yes, my bad for not remembering that and wasting time.  But the other three are really appropriate to be just simple triggers and can stay like that.  There is no other code needed.

Applications really need something like a MenuButton that takes arbitrary menu items and does the work for you of placement etc.!  This is similar to our SplitButton project on Desktop, and could perhaps be folded in together in terms of adding something with consistent naming.

**Comment by TomTirapani** [comment]
Definitely agree that a component that does the work of creating and placing the menu for you would be much better - I'll see what can be done!

**Comment by amcclain** [comment]
Is this all specific to mobile?  Note https://github.com/xh/hoist-react/pull/1800 for earlier discussion from last spring about building a desktop `MenuButton` (as a more general / first-order add to support notion of a `SplitButton`).

**Comment by lbwexler** [comment]
Tom and I discussed this in-detail, and I gave him a few minor suggestions.  Some big points we agreed on:

+ We are going to have to live with the bifurcation between mobile/desktop menus and menuitems for a while, maybe long-term.  But this change will really make a big step in the right direction.

+ We need general popover like support on mobile to support apps like the big client mobile app we are currently working on. This is a seperate (but critical) project, and Tom is going to make a ticket for that.
